packages/rabbitmq/src/amqp/connection.ts

```diff
@@ -77,6 +100,9 @@ export class AmqpConnection {
   private _channel!: Channel;
   private _channels: Record<string, Channel> = {};
   private _connection?: Connection;
+
+  private _consumers: ConsumerHandler<unknown, unknown>[] = [];
```

Comment:
Can we switch using a `Map<string, ConsumerHandler>` instead and use the `consumerTag` as the key? It would simplify a lot of the other code so that there isn't a need to do things like `findIndex` and `splice`